cheetah/accelerator/dipole.py

```diff
@@ -1,47 +1,69 @@
-from typing import Optional, Union
+from typing import Literal, Optional, Union
 
 import matplotlib.pyplot as plt
 import numpy as np
 import torch
 from matplotlib.patches import Rectangle
+from scipy.constants import physical_constants
 from torch import Size, nn
 
+from cheetah.particles import Beam, ParticleBeam
 from cheetah.track_methods import base_rmatrix, rotation_matrix
-from cheetah.utils import UniqueNameGenerator
+from cheetah.utils import UniqueNameGenerator, bmadx
 
 from .element import Element
 
 generate_unique_name = UniqueNameGenerator(prefix="unnamed_element")
 
+electron_mass_eV = torch.tensor(
+    physical_constants["electron mass energy equivalent in MeV"][0] * 1e6
+)
+
 
 class Dipole(Element):
     """
     Dipole magnet (by default a sector bending magnet).
 
     :param length: Length in meters.
+    :param p0c: Reference momentum at dipole in eV/c.
     :param angle: Deflection angle in rad.
     :param k1: Focussing strength in 1/m^-2.
     :param e1: The angle of inclination of the entrance face [rad].
     :param e2: The angle of inclination of the exit face [rad].
     :param tilt: Tilt of the magnet in x-y plane [rad].
+    :param gap: The magnet gap in meters. Note that in MAD and ELEGANT: HGAP = gap/2.
+    :param gap_exit: The magnet gap at the entrance in meters. Note that in MAD and
+        ELEGANT: HGAP = gap/2. Only set if different from `gap`.
     :param fringe_integral: Fringe field integral (of the enterance face).
-    :param fringe_integral_exit: (only set if different from `fint`) Fringe field
-        integral of the exit face.
-    :param gap: The magnet gap [m], NOTE in MAD and ELEGANT: HGAP = gap/2
+    :param fringe_integral_exit: Fringe field integral of the exit face. Only set if
+        different from `fringe_integral`.
+    :param fringe_at: Where to apply the fringe fields. The available options are:
+        - "both_ends": Apply fringe fields at both ends.
+        - "entrance_end": Apply fringe fields at the entrance end.
+        - "exit_end": Apply fringe fields at the exit end.
+        - "no_end": Do not apply fringe fields.
+    :param fringe_type: Type of fringe field. Currently only supports `"linear_edge"`.
     :param name: Unique identifier of the element.
```

Comment:
@jp-ga Do any of these only apply for `bmadx` tracking? If so, I think this should be mentioned in the docstring.